src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -4462,11 +4462,22 @@ MergeTaskList(MapMergeJob *mapMergeJob, List *mapTaskList, uint32 taskIdIndex)
 		{
 			Task *mapTask = (Task *) lfirst(mapTaskCell);
 
-			/* we need node names for the query, and we'll resolve them later */
-			char *undefinedQueryString = NULL;
+			/* find the node name/port for map task's execution */
+			List *mapTaskPlacementList = mapTask->taskPlacementList;
+
+			ShardPlacement *mapTaskPlacement = linitial(mapTaskPlacementList);
```

Comment:
This could use more elaboration. Specifically that we're optimistically assuming that we'll be able to use the first placement. In the task-tracker executor we may fail over to a different placement and rebuild the query string.